website/build_site.sh

```diff
@@ -1,37 +1,110 @@
-#!/bin/sh
+#!/bin/bash
+#
+# A simple script to build the github IO pages out of
+
+# be strict
+set -eu
+
+# be verbose
+#set -x
 
 # ensure we're in the right folder
 scriptdir=$(readlink -f "$(dirname "$0")")
-cd "$scriptdir"
+MLOS_ROOT=$(readlink -f "$scriptdir/..")
+cd "$MLOS_ROOT/website"
 
-cp -r ../documentation content/
-# downgrade html output because hugo doesn't like raw html
-jupyter nbconvert ../source/Mlos.Notebooks/*.ipynb --to markdown --output-dir content/notebooks --template nbconvert_template.md.j2 --config jupyter_nbconvert_config.py
+. "$MLOS_ROOT/scripts/util.sh"
+pythonCmd=$(getPythonCmd)
 
-# nbconvert and hugo disagree about paths
-# this should probably be done via the template
-sed -i 's/BayesianOptimization_files/\.\.\/BayesianOptimization_files/g' content/notebooks/BayesianOptimization.md
-sed -i 's/SmartCacheOptimization_files/\.\.\/SmartCacheOptimization_files/g' content/notebooks/SmartCacheOptimization.md
+# Make sure we have all the requirements for generating notebooks.
+$pythonCmd -m pip install -e "$MLOS_ROOT/source/Mlos.Python/"
+$pythonCmd -m pip install pyyaml
+$pythonCmd -m pip install \
+    $("$MLOS_ROOT/scripts/parse-pip-requirements-from-environment-yaml.py" "$MLOS_ROOT/source/Mlos.Notebooks/environment.yml")
+$pythonCmd -m pip install jupyter nbconvert
+
+# execute and render the notebooks to html
+# downgrade html output because hugo doesn't like raw html
+mkdir -p content/notebooks
+notebooks='BayesianOptimization SmartCacheOptimization'
+for nb in $notebooks; do
+    nb_path="$MLOS_ROOT/source/Mlos.Notebooks/$nb.ipynb"
+    echo "Executing and rendering $nb_path"
+    $pythonCmd -m jupyter nbconvert "$nb_path" \
+        --to markdown --output-dir content/notebooks/ \
+        --template nbconvert_template.md.j2 --config jupyter_nbconvert_config.py
+# TODO: Execute the notebooks during website build.
+# Currently the SmartCacheOptimization throws an error.
+# NOTE: These are also somewhat expensive to execute on every single CI.
+#        --execute \
+#        --ExecutePreprocessor.kernel_name=python3 \
+#        --ExecutePreprocessor.timeout=600 \
```

Comment:
These currently fail.  I haven't debugged it yet, but looked to be similar to the `Shell=True` fixup from #46

Reply:
The intent here was to handle #55 while I was at it.